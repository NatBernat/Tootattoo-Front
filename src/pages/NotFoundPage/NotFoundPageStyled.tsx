import styled from "styled-components";

const NotFoundPageStyled = styled.div`
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);

  display: flex;
  flex-direction: column;
  gap: 20px;
  padding: 20px;
  justify-content: center;
  align-items: center;
  text-align: center;
  background-color: #ffbfbf;
  border-radius: 8px;
  img {
    height: 100px;
    width: auto;
  }
  * {
    margin: 0;
  }
`;

export default NotFoundPageStyled;
