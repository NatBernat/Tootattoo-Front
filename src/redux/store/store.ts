import { configureStore } from "@reduxjs/toolkit";
import usersReducer from "../features/userSlice/userSlice";
import tattoosReducer from "../features/tattoosSlice/tattoosSlice";

const store = configureStore({
  reducer: {
    user: usersReducer,
    tattoos: tattoosReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
