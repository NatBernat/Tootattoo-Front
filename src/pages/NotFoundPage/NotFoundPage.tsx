import NotFoundPageStyled from "./NotFoundPageStyled";

const NotFoundPage = (): JSX.Element => {
  return (
    <NotFoundPageStyled className="page">
      <h2>Page not found</h2>
      <p>The page you're looking for doesn't exist.</p>
      <img src="./images/imageLogo.svg" alt="tootattoo logo" />
    </NotFoundPageStyled>
  );
};

export default NotFoundPage;
