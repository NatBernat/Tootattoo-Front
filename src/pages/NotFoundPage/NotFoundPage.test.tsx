import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../redux/store/store";
import NotFoundPage from "./NotFoundPage";

describe("Given NotFoundPage component", () => {
  describe("When it's instantiated", () => {
    test("Then it should render a paragraph", () => {
      const expectedText = "Page not found";

      render(
        <Provider store={store}>
          <NotFoundPage />
        </Provider>
      );

      const testedParagraph = screen.getByText(expectedText);

      expect(testedParagraph).toBeInTheDocument();
    });
  });
});
