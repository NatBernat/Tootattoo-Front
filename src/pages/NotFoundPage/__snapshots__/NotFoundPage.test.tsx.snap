// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given NotFoundPage component When it's rendered Then it should always match this snapshot 1`] = `
<div
  className="sc-bczRLJ knOfCy page not-found-page"
>
  <h2>
    Page not found
  </h2>
  <img
    alt="tootattoo logo"
    src="./images/imageLogo.svg"
  />
</div>
`;
